### PR TITLE
Adding support for quote wrapped windows event channels

### PR DESCRIPTION
**Pull request by poppyseedplehzr**
This checks for, and removes, quotes wrapping the Windows event log channel subscriptions. Now one can pass a coma delimited list, where channel names contain spaces, and they'll correctly be subscribed to.  Example output follows:
```
C:\Users\thor\work\repos\osquery [win-events-channel-quotes]> .\build\windows10\osquery\Release\osqueryi.exe --disable_events=false --windows_event_channels="System,Security,Windows PowerShell" --verbose
I0418 15:10:02.155122  7256 process_ops.cpp:162] Unable to find environment variable (0): OSQUERY_WORKER
I0418 15:10:02.161072  7256 init.cpp:380] osquery initialized [version=2.4.0-15-gef1f3975]
I0418 15:10:02.161072  7256 process_ops.cpp:162] Unable to find environment variable (203): OSQUERY_WORKER
I0418 15:10:02.161072  7256 extensions.cpp:295] Could not autoload extensions: Failed reading: \ProgramData\osquery\extensions.load
I0418 15:10:02.163064  7256 process_ops.cpp:162] Unable to find environment variable (203): OSQUERY_EXTENSIONS
I0418 15:10:02.163064  7256 extensions.cpp:301] Windows does not support loadable modules
I0418 15:10:02.163064  6104 interface.cpp:317] Extension manager service starting: \\.\pipe\shell.em
I0418 15:10:02.163064  7256 init.cpp:615] Error reading config: config file does not exist: \ProgramData\osquery\osquery.conf
I0418 15:10:02.167075  4576 events.cpp:748] Starting event publisher run loop: windows_event_log
I0418 15:10:02.167075  7256 process_ops.cpp:162] Unable to find environment variable (203): TERM
Using a virtual database. Need help, type '.help'
osquery> select * from windows_events where source="Windows PowerShell";
I0418 15:10:10.653786  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.653786  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.653786  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.655553  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.655553  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.655553  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.655553  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.657544  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.657544  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.657544  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 15:10:10.657544  7256 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
+------------+--------------------------------+--------------------+---------------+---------------+---------+------+-------+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| time       | datetime                       | source             | provider_name | provider_guid | eventid | task | level | keywords | data
                                                                                                                                                                                                                                                                                                             |
+------------+--------------------------------+--------------------+---------------+---------------+---------+------+-------+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1492553407 | 2017-04-18T22:10:07.468019900Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Registry\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=1\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=e1399cca-ce07-42ec-aac2-b80aaf14979b\r\n\tHostApplication=C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                       |
| 1492553407 | 2017-04-18T22:10:07.468019900Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Alias\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=3\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=e1399cca-ce07-42ec-aac2-b80aaf14979b\r\n\tHostApplication=C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                          |
| 1492553407 | 2017-04-18T22:10:07.468019900Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Environment\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=5\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=e1399cca-ce07-42ec-aac2-b80aaf14979b\r\n\tHostApplication=C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                    |
| 1492553407 | 2017-04-18T22:10:07.470014000Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=FileSystem\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=7\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=e1399cca-ce07-42ec-aac2-b80aaf14979b\r\n\tHostApplication=C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                     |
| 1492553407 | 2017-04-18T22:10:07.470014000Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Function\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=9\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=e1399cca-ce07-42ec-aac2-b80aaf14979b\r\n\tHostApplication=C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                       |
| 1492553407 | 2017-04-18T22:10:07.470014000Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Variable\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=11\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=e1399cca-ce07-42ec-aac2-b80aaf14979b\r\n\tHostApplication=C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                      |
| 1492553407 | 2017-04-18T22:10:07.480059100Z | Windows PowerShell | PowerShell    |               | 400     | 4    | 4     | -1       | {
    "EventData": {
        "Data": "\tNewEngineState=Available\r\n\tPreviousEngineState=None\r\n\r\n\tSequenceNumber=13\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=e1399cca-ce07-42ec-aac2-b80aaf14979b\r\n\tHostApplication=C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=5.1.14393.1066\r\n\tRunspaceId=e9600d22-0d0f-4f20-b2c7-ecaecb3fbbbd\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
 |
+------------+--------------------------------+--------------------+---------------+---------------+---------+------+-------+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

### CODE REVIEW

**Comment by poppyseedplehzr** [comment]
This fixes #3173

File: osquery/tables/events/windows/windows_events.cpp [hunk 1]
```diff
@@ -42,7 +42,12 @@ class WindowsEventSubscriber
  public:
   Status init() override {
     auto wc = createSubscriptionContext();
-    for (const auto& chan : osquery::split(FLAGS_windows_event_channels, ",")) {
+    for (auto& chan : osquery::split(FLAGS_windows_event_channels, ",")) {
+      // We remove quotes if they exist
+      auto quoteLoc = chan.find("\"");
+      if (quoteLoc != std::string::npos) {
+        chan.erase(quoteLoc);
```
Comment (jedi22) [inline osquery/tables/events/windows/windows_events.cpp]: Are any of the things below valid?
`--windows_event_channels=System,Security,"Windows PowerShell"`
`--windows_event_channels='System,Security,Windows PowerShell'`
`--windows_event_channels=System,Security,'Windows PowerShell'`
`--windows_event_channels="System,Security,'Windows PowerShell'"`

Comment (poppyseedplehzr) [inline osquery/tables/events/windows/windows_events.cpp]: Good call. I think those all might break it. The objective is to get rid of all quotes. Thanks dude!

**Comment by poppyseedplehzr** [comment]
Updated testing output:
```
C:\Users\Nicholas\projects\repos\osquery [win-events-channel-quotes ≡ +0 ~1 -0 !]> .\build\windows10\osquery\Release\osqueryi.exe --disable_events=false --windows_event_channels="System,Security,'Windows PowerShell'" "select * from osquery_flags where name like '%windows_event_channels%';"
I0418 19:50:00.815892 13436 init.cpp:615] Error reading config: config file does not exist: \ProgramData\osquery\osquery.conf
+------------------------+--------+----------------------------------------------------+-----------------------------------+--------------------------------------+------------+
| name                   | type   | description                                        | default_value                     | value                                | shell_only |
+------------------------+--------+----------------------------------------------------+-----------------------------------+--------------------------------------+------------+
| windows_event_channels | string | Comma-separated list of Windows event log channels | System,Application,Setup,Security | System,Security,'Windows PowerShell' | 0          |
+------------------------+--------+----------------------------------------------------+-----------------------------------+--------------------------------------+------------+
C:\Users\Nicholas\projects\repos\osquery [win-events-channel-quotes ≡ +0 ~1 -0 !]> .\build\windows10\osquery\Release\osqueryi.exe --disable_events=false --windows_event_channels=System,Security,'Windows PowerShell' "select * from osquery_flags where name like '%windows_event_channels%';"
I0418 19:50:04.945688 20080 init.cpp:615] Error reading config: config file does not exist: \ProgramData\osquery\osquery.conf
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
| name                   | type   | description                                        | default_value                     | value                              | shell_only |
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
| windows_event_channels | string | Comma-separated list of Windows event log channels | System,Application,Setup,Security | System,Security,Windows PowerShell | 0          |
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
C:\Users\Nicholas\projects\repos\osquery [win-events-channel-quotes ≡ +0 ~1 -0 !]> .\build\windows10\osquery\Release\osqueryi.exe --disable_events=false --windows_event_channels='System,Security,Windows PowerShell' "select * from osquery_flags where name like '%windows_event_channels%';"
I0418 19:50:10.655943  6024 init.cpp:615] Error reading config: config file does not exist: \ProgramData\osquery\osquery.conf
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
| name                   | type   | description                                        | default_value                     | value                              | shell_only |
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
| windows_event_channels | string | Comma-separated list of Windows event log channels | System,Application,Setup,Security | System,Security,Windows PowerShell | 0          |
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
C:\Users\Nicholas\projects\repos\osquery [win-events-channel-quotes ≡ +0 ~1 -0 !]> .\build\windows10\osquery\Release\osqueryi.exe --disable_events=false --windows_event_channels=System,Security,"Windows PowerShell" "select * from osquery_flags where name like '%windows_event_channels%';"
I0418 19:50:22.647994  8280 init.cpp:615] Error reading config: config file does not exist: \ProgramData\osquery\osquery.conf
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
| name                   | type   | description                                        | default_value                     | value                              | shell_only |
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
| windows_event_channels | string | Comma-separated list of Windows event log channels | System,Application,Setup,Security | System,Security,Windows PowerShell | 0          |
+------------------------+--------+----------------------------------------------------+-----------------------------------+------------------------------------+------------+
```

**Comment by poppyseedplehzr** [comment]
I was a little concerned with the `--windows_event_channels="System,Security,'Windows PowerShell'"` case, but it looks to be registering just fine:
```
osquery> select * from windows_events where source='Windows PowerShell';
I0418 19:52:36.946810  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.946810  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.946810  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.946810  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.946810  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.946810  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.947813  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.947813  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.947813  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.947813  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
I0418 19:52:36.947813  3952 process_ops.cpp:162] Unable to find environment variable (203): ENHANCE
+------------+--------------------------------+--------------------+---------------+---------------+---------+------+-------+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| time       | datetime                       | source             | provider_name | provider_guid | eventid | task | level | keywords | data

                                                                                                                                                                                   |
+------------+--------------------------------+--------------------+---------------+---------------+---------+------+-------+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 1492570342 | 2017-04-19T02:52:22.111796400Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Registry\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=1\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=1bd9766a-3622-4749-bd3b-da6912bb2413\r\n\tHostApplication=C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                       |
| 1492570342 | 2017-04-19T02:52:22.112798500Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Alias\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=3\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=1bd9766a-3622-4749-bd3b-da6912bb2413\r\n\tHostApplication=C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                          |
| 1492570342 | 2017-04-19T02:52:22.112798500Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Environment\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=5\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=1bd9766a-3622-4749-bd3b-da6912bb2413\r\n\tHostApplication=C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                    |
| 1492570342 | 2017-04-19T02:52:22.114803900Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=FileSystem\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=7\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=1bd9766a-3622-4749-bd3b-da6912bb2413\r\n\tHostApplication=C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                     |
| 1492570342 | 2017-04-19T02:52:22.114803900Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Function\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=9\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=1bd9766a-3622-4749-bd3b-da6912bb2413\r\n\tHostApplication=C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                       |
| 1492570342 | 2017-04-19T02:52:22.114803900Z | Windows PowerShell | PowerShell    |               | 600     | 6    | 4     | -1       | {
    "EventData": {
        "Data": "\tProviderName=Variable\r\n\tNewProviderState=Started\r\n\r\n\tSequenceNumber=11\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=1bd9766a-3622-4749-bd3b-da6912bb2413\r\n\tHostApplication=C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=\r\n\tRunspaceId=\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
                                                      |
| 1492570342 | 2017-04-19T02:52:22.120819900Z | Windows PowerShell | PowerShell    |               | 400     | 4    | 4     | -1       | {
    "EventData": {
        "Data": "\tNewEngineState=Available\r\n\tPreviousEngineState=None\r\n\r\n\tSequenceNumber=13\r\n\r\n\tHostName=ConsoleHost\r\n\tHostVersion=5.1.14393.1066\r\n\tHostId=1bd9766a-3622-4749-bd3b-da6912bb2413\r\n\tHostApplication=C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n\tEngineVersion=5.1.14393.1066\r\n\tRunspaceId=1cb05d18-1a80-4627-9716-960b61b13b65\r\n\tPipelineId=\r\n\tCommandName=\r\n\tCommandType=\r\n\tScriptName=\r\n\tCommandPath=\r\n\tCommandLine="
    }
}
 |
+------------+--------------------------------+--------------------+---------------+---------------+---------+------+-------+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by osqueryer** [comment]
:-1: The commit 50bd23fb94f6eefb72d1133b5029c4b5024a75dd (Job results: [4487](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4487/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 50bd23fb94f6eefb72d1133b5029c4b5024a75dd (Job results: [4488](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4488/)) failed one or more tests (**Linux**).

**Comment by poppyseedplehzr** [comment]
ok to test

**Comment by osqueryer** [comment]
:-1: The commit 50bd23fb94f6eefb72d1133b5029c4b5024a75dd (Job results: [1398](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1398/)) failed one or more tests (**Windows**).

**Comment by poppyseedplehzr** [comment]
Ok to test

**Pull request merged**